pkg/controllers/plugin/plugin_workload_status_controller.go

```diff
@@ -110,26 +143,26 @@ func (r *WorkLoadStatusReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		releaseStatus.ReleaseNamespace = helmRelease.Namespace
 		releaseStatus.ClusterName = plugin.Spec.ClusterName
 		releaseStatus.HelmStatus = helmRelease.Info.Status.String()
-		objectMap, _ := helm.ObjectMapFromManifest(restClientGetter, plugin.Namespace, helmRelease.Manifest, &helm.ManifestObjectFilter{})
-
-		for key := range objectMap {
-			if key.GVK.Version == "v1" &&
-				(key.GVK.Group == "apps" || key.GVK.Group == "batch") ||
-				(key.GVK.Group == monitoringAPIGroupName && key.GVK.Kind == "Alertmanager") {
-				getStatusPayload(ctx, releaseStatus, objClient, key.Name, releaseStatus.ReleaseNamespace, key.GVK)
-			}
+		fileteredObjectMap, err := helm.ObjectMapFromManifestWithMultipleFilters(restClientGetter, plugin.Namespace, helmRelease.Manifest, &helm.ManifestMultipleObjectFilter{
+			Filters: objectFilter,
+		})
+		if err != nil {
+			log.FromContext(ctx).Error(err, "failed to get object map from manifest")
+		}
+		for _, key := range fileteredObjectMap {
+			getPayloadStatus(ctx, releaseStatus, objClient, key.ObjectKey.Name, releaseStatus.ReleaseNamespace, key.ObjectKey.GVK)
 		}
 
 		if statusErr := r.setStatus(ctx, plugin, releaseStatus, pluginStatus); statusErr != nil {
 			log.FromContext(ctx).Error(statusErr, "failed to set status")
 		}
```

Comment:
The `pluginStatus` is not updated. The `status` created within `getPayloadStatus(..)` is populated within the function but not returned/ set on the pluginStatus.

Reply:
done in [7b37d0b](https://github.com/cloudoperators/greenhouse/pull/398/commits/7b37d0b7e294576e528b54fab8babb50a2b9dcfb)